### ISSUE TITLE
Water rendering issue in certain places

### ISSUE THREAD

**Issue by dmklinger** [osmandapp/OsmAnd#19141]
### Description

In certain places, water is improperly rendered on land. 

I don't know exactly how your rendering engine works, but it looks to me like you have some sort of optimization where tiles that are all water get their background color changed to the water color rather than having to render a huge polygon, but some tiles are incorrectly labeled as water.

The specific tiles that are affected appears to be dependent on zoom levels.

### Steps to reproduce

Exact locations:

1) Istanbul: https://www.openstreetmap.org/#map=16/41.0662/29.0130

2) Algiers: https://www.openstreetmap.org/#map=15/36.7215/3.1366

### Actual result

Example 1) Istanbul:

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/25443497/3838eb1e-4e01-48ea-aa42-26b090e0714b"  width="300" height="auto"/> </th>
  </tr>
</table>

And after zooming in, the same bug exists but the tiles are different:

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/25443497/8b2bdd75-cace-4c80-8886-53827d3aea39"  width="300" height="auto"/> </th>
  </tr>
</table>

Example 2) Algiers:
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/25443497/0d2c0b22-da48-4a17-ada9-0316bbaf6bd5"  width="300" height="auto"/> </th>
  </tr>
</table>

### Expected result

The background should be grey, not blue.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: 4.6.6.1
Android/iOS version: iPadOS
Device model: 4th gen 11inch pro
Crash-Logs: n/a
```

**Comment by snooppr** [comment]
Duplicate, same problem: https://github.com/osmandapp/OsmAnd/issues/19133

**Comment by yuriiurshuliak** [comment]
The bug was not reproduced. I made an attempt to reproduce the flooding issue in Istanbul using two render versions, but the map rendered without any signs of flooding. Please try to update the map.
Regarding the flooding in El Harrach, Algeria, this refers to #19035.

OsmAnd~ 4.7.0#1634m, released: 2024-02-20

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/127092082/c18fc17f-7c37-46fb-86b9-d5492ef3b287"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/127092082/de8c877b-952a-4d5d-9a2e-1a12643e9e56"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by dmklinger** [comment]
Interesting. I'm guessing it's an iOS specific bug. I uninstalled and reinstalled the map and still had the same issue. Additionally, I installed the map on my iPhone and got the same result

![image](https://github.com/osmandapp/OsmAnd/assets/25443497/e15c3022-9998-4c22-a7bf-0e1f2bd04ef7)

(all default settings here)

**Comment by dmpr0** [comment]
I can reproduce issue, both on Android Google Pixel 6A, and iOS on iPhone XS Max.

<table style="width:100%">
  <tr>
    <th colspan=2>Android</th>
    <th colspan=2>iOS</th>
  </tr>
  <tr>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/9365308/a8c5f9f3-9a9e-4a44-b846-f28a1721092d"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/9365308/7376f9bf-06d2-4f60-bca0-89fc1bea4af7"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/9365308/0cca6574-313c-4843-841f-ab91f55cdc85"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/9365308/2e60a9c5-5a3b-4b58-9fa0-0fb01ce1d62b"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by xmd5a2** [comment]
**Case 1:** basemap coastline is precise enough and correct.
![Screenshot_2024-02-28_15-00-52](https://github.com/osmandapp/OsmAnd/assets/5829014/e82c02df-f186-4a6b-bd33-d207b460fb2d)
Coastline in OSM are correct also. Reason of flooding is not clear. Reproduced in OsmAndMapCreator OpenGL.

**Case 2:** issue was in incorrect coastline "island" inside mainland. This is an error and it is known to cause flooding in OsmAnd. I fixed this in OSM. New maps will be extracted and regenerated 1-2 march. There is no reason to extract them manually in the end of the month.
![Screenshot_2024-02-28_15-10-02](https://github.com/osmandapp/OsmAnd/assets/5829014/a1ce9d1d-6196-47c8-90fd-fe3b0044acdd)
Map (OpenGL) after fix (generated manually):
![изображение](https://github.com/osmandapp/OsmAnd/assets/5829014/b7e22bc2-55cc-4c65-88cb-84806bf037c1)

**Comment by zipav** [comment]
OsmAnd Version: OsmAnd Nightly 4.7.0#1872m, released: 2024-03-19/ OsmAnd TestFlight 4.7.2.10
Android/iOS version: Android 12/iOS 16.0
Device model: Nokia 5.3/ iPhone 12


<table style="width:100%">
  <tr>
    <th colspan=2>Android</th>
    <th colspan=2>iOS</th>
  </tr>
  <tr>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/123936521/634c07ac-f910-4804-9bdf-9513858aa314"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/123936521/42e7a563-0210-480f-98cc-8adafc95c6af"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/123936521/0f940bd0-ad78-4105-bb7d-a959e50d2ebf"  width="300" height="auto"/> </td>
    <td><img src="https://github.com/osmandapp/OsmAnd/assets/123936521/7895f10b-9f04-4ced-8cdd-f717237cd8f9"  width="300" height="auto"/> </td>
  </tr>
</table>
